qiita_db/analysis.py

```diff
@@ -266,6 +275,21 @@ def name(self):
         return conn_handler.execute_fetchone(sql, (self._id, ))[0]
 
     @property
+    def portal(self):
```

Comment:
I know how it works, but it is unexpected behavior as the instance is
inconsistent with the definition and the name change is implicit. Since the
purpose of this method is debug, flag it as such in the doc string or even
the function name if you want to be explicit (eg `_DEBUG_foo`. Using a
double underscore will only confuse a developer, which is the last thing
you want to do when you need to urgently use debug methods.

On Tue, Jun 16, 2015 at 4:46 PM, josenavas <notifications@github.com> wrote:

> In qiita_db/analysis.py
> <https://github.com/biocore/qiita/pull/1256#discussion_r32579924>:
>
> > @@ -266,6 +275,21 @@ def name(self):
> >          return conn_handler.execute_fetchone(sql, (self._id, ))[0]
> >
> >      @property
> > +    def portal(self):
>
> That is not unexpected. Accordingly to the python documentation
> <https://docs.python.org/2/tutorial/classes.html#private-variables-and-class-local-references>
> :
>
> Any identifier of the form __spam (at least two leading underscores, at
> most one trailing underscore) is textually replaced with _classname__spam
>
> In [21]: class foo(object):
>    ....:     def __bar(self):
>    ....:         print self
>    ....:
>
> In [22]: x = foo()
>
> In [23]: x.__bar()---------------------------------------------------------------------------AttributeError                            Traceback (most recent call last)<ipython-input-23-883553e2ea7d> in <module>()----> 1 x.__bar()
> AttributeError: 'foo' object has no attribute '__bar'
>
> In [24]: x._foo__bar()<__main__.foo object at 0x103f9d1d0>
>
> which I think is what we want. Adding a check is not ok as it will not
> allow us to use it to debug the live system.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiita/pull/1256/files#r32579924>.
>